src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/private/aligned_storage.hh

```diff
@@ -0,0 +1,52 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/v1/detail/macros.hpp>
+
+#include <memory>
+
+namespace bsoncxx {
+
+// Workaround lack of aligned allocation support prior to C++17.
+//
+// Unlike `std::aligned_storage` and `alignas`, this type does NOT propagate alignment requirements to the parent
+// object. This permits correct behavior despite use with regular (alignment-unaware) allocation functions.
+//
+// `alignof(T)` must be explicitly provided by the user for correct behavior, as template parameters do not necessarily
+// inherit attributes such as alignment, e.g. `std::size_t alignment = alignof(T)` does not work as expected.
+template <std::size_t size, std::size_t alignment>
+class aligned_storage {
+    // Sanity check.
+    static_assert(size >= alignment, "sizeof(T) >= alignof(T) must be true");
+
+   private:
+    unsigned char _storage[2u * size];
```

Comment:
Suggest using `alignment + size` rather than `2u * size`. I expect the worst case to align to an arbitrary address only requires an extra `alignment` bytes.

Reply:
Good call. Updated accordingly + added a comment to document rationale.